apidoc/Titanium/UI/Matrix2D.yml

```diff
@@ -0,0 +1,279 @@
+            of an animation.
+        type: Number
+
+      - name: sy
+        summary: |
+            Vertical scaling factor. If `toSx` and `toSy` are specified,
+            this specifies the starting vertical scaling factor, at the beginning of
+            an animation.
+        type: Number
+
+      - name: toSx
+        summary: |
+            Ending horizontal scaling factor, at the end of an animation.
+            If specified, `toSy` must be specified as well. Android only.
+        type: Number
+        optional: true
+
+      - name: toSy
+        summary: |
+            Ending vertical scaling factor, at the end of an animation.
+            If specified, `toSx` must be specified as well. Android only.
+        type: Number
+        optional: true
+
+
+  - name: translate
+    summary: Returns a matrix constructed by applying a translation transform to this matrix.
+    returns:
+        type: Titanium.UI.Matrix2D
+    parameters:
+      - name: tx
+        summary: Horizontal component of the translation.
+        type: Number
+      - name: ty
+        summary: Vertical component of the translation.
+        type: Number
+properties:
+  - name: a
+    summary: The entry at position [1,1] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+  - name: b
+    summary: The entry at position [1,2] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+  - name: c
+    summary: The entry at position [2,1] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+  - name: d
+    summary: The entry at position [2,2] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+  - name: tx
+    summary: The entry at position [3,1] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+  - name: ty
+    summary: The entry at position [3,2] in the matrix.
+    type: Number
+    exclude-platforms: [android]
+
+examples:
+  - title: Apply a 2D Matrix to a Label
+    example: |
+        The following uses a 2D matrix to translate a label in the y direction.
+
+            var win = Ti.UI.createWindow({
+              backgroundColor: 'white'
+            });
+
+            var label = Ti.UI.createLabel({
+              font:{fontSize:50},
+              text:'Titanium',
+              textAlign:'center',
+              top: 100
+            });
+            win.add(label);
+
+            var button = Ti.UI.createButton({
+              title:'Animate',
+              bottom:20,
+              width:200, height:40
+            });
+            win.add(button);
+
+            button.addEventListener('click', function(){
+              var t1 = Ti.UI.createMatrix2D();
+              t1 = t1.translate(0, 300);
+              var a1 = Ti.UI.createAnimation();
+              a1.transform = t1;
+              a1.duration = 800;
+              label.animate(a1);
+            });
+            win.open();
+
+---
+name: MatrixCreationDict
```

Comment:
Since this type is now here and under the same name in [2DMatrix.yml](../../tree/master/apidoc/Titanium/UI/2DMatrix.yml) it causes duplicates in the generated docs. The docs script is smart enough to detect the duplicate and tries to merge them but that generates duplicate property entries ([see common.js](../../tree/master/apidoc/lib/common.js#L189)).

Reply:
@janvennemann I have updated our script to actually throw an error if duplicate keys are used.